src/django_components/component.py

```diff
@@ -538,6 +541,11 @@ def render(
           is rendered. The keys on the context can be accessed from within the template.
             - NOTE: In "isolated" mode, context is NOT accessible, and data MUST be passed via
               component's args and kwargs.
+        - `kind` - Configure how to handle JS and CSS dependencies.
```

Comment:
Another difference from the tech doc - in the tech doc I suggested to add kwarg named `render_dependencies` to `Component.render()` and `Component.render_to_response()`.

During the preparation of the MR, I realized that a better name would be if the kwarg reflected the intent.

So using `kind="document"` means "process the rendered string as HTML document", and it is the default behaviour.

```py
Component.render(kind="document")
```

This way down the line it will be possible to add `kind="fragment"` to render the JS / CSS dependencies for the HTML fragment use case.

```py
Component.render(kind="fragment")
```

In this MR one can also pass `kind=None`, which is [currently a workaround as described here](https://github.com/EmilStenstrom/django-components/pull/688/files#r1784292398), and hopefully I will be able to remove it in one of the following MRs.

Reply:
Turns out that `as` is a reserved keyword :D 

https://realpython.com/lessons/reserved-keywords/